e2e/Pages/MapComponent.ts

```diff
@@ -0,0 +1,120 @@
+  readonly adminBoundry: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.page = page;
+    this.mapComponent = this.page.getByTestId('dashboard-map-componenet');
+    this.breadCrumbNationalView = this.page.getByTestId(
+      'breadcrumb-national-view',
+    );
+    this.breadCrumbEventView = this.page.getByTestId('breadcrumb-event-view');
+    this.breadCrumbAdminAreaView = this.page.getByTestId(
+      'breadcrumb-admin-area-view',
+    );
+    this.breadCrumbAdminArea2View = this.page.getByTestId(
+      'breadcrumb-admin-area-2-view',
+    );
+    this.breadCrumbAdminArea3View = this.page.getByTestId(
+      'breadcrumb-admin-area-3-view',
+    );
+    this.legend = this.page.getByTestId('map-legend');
+    this.layerMenu = this.page.getByTestId('layer-menu');
+    this.adminBoundry = this.page.locator('.leaflet-interactive');
+  }
+
+  async mapComponentIsVisible() {
+    await expect(this.mapComponent).toBeVisible();
+  }
+
+  async breadCrumbViewIsVisible({
+    nationalView = false,
+    eventView = false,
+    adminAreaView = false,
+    adminAreaView2 = false,
+    adminAreaView3 = false,
+  }: {
+    nationalView?: boolean;
+    eventView?: boolean;
+    adminAreaView?: boolean;
+    adminAreaView2?: boolean;
+    adminAreaView3?: boolean;
+  }) {
+    if (nationalView) {
+      await expect(this.breadCrumbNationalView).toBeVisible();
+    } else {
+      await expect(this.breadCrumbNationalView).toBeHidden();
+    }
+    if (eventView) {
+      await expect(this.breadCrumbEventView).toBeVisible();
+    } else {
+      await expect(this.breadCrumbEventView).toBeHidden();
+    }
+    if (adminAreaView) {
+      await expect(this.breadCrumbAdminAreaView).toBeVisible();
+    } else {
+      await expect(this.breadCrumbAdminAreaView).toBeHidden();
+    }
+    if (adminAreaView2) {
+      await expect(this.breadCrumbAdminArea2View).toBeVisible();
+    } else {
+      await expect(this.breadCrumbAdminArea2View).toBeHidden();
+    }
+    if (adminAreaView3) {
+      await expect(this.breadCrumbAdminArea3View).toBeVisible();
+    } else {
+      await expect(this.breadCrumbAdminArea3View).toBeHidden();
+    }
+  }
+
+  async isLegendOpen({ legendOpen = false }: { legendOpen?: boolean }) {
+    if (legendOpen) {
+      await expect(this.legend).toHaveAttribute('open');
+    } else {
+      await expect(this.legend).not.toHaveAttribute('open');
+    }
+  }
+
+  async isLayerMenuOpen({
+    layerMenuOpen = false,
+  }: {
+    layerMenuOpen?: boolean;
+  }) {
+    if (layerMenuOpen) {
+      await expect(this.layerMenu).toBeVisible();
+    } else {
+      await expect(this.layerMenu).toBeHidden();
+    }
+  }
+
+  async assertAdminBoundariesVisible() {
+    await this.page.waitForLoadState('networkidle');
+    await this.page.waitForLoadState('domcontentloaded');
+    // await this.page.waitForTimeout(500); // This is a workaround for the issue with the map not loading in the same moment as the dom
```

Comment:
Leaflet is weird this way. I have to check if they've become more predictable in the last 3 years. This is okay for now. Why is it commented out?

Reply:
I do not want to keep hardcoded waits. It is more reliable if Playwright waits explicitly for a state or element then to stop the whole execution. In the long run it also makes test execution faster. I kept it there as a point of reference if nothing else works.